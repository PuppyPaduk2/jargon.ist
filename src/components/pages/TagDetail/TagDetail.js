// @flow

import React, { Fragment } from 'react';
import Link from 'gatsby-link';

import { type Jargon } from '../../../types/jargon';

type Props = {
  tag: string,
  jargons: Array<Jargon>,
  totalCount: number,
};

const TagDetail = ({ tag, jargons, totalCount }: Props) => (
  <Fragment>
    <h1>
      {tag} ({totalCount})
    </h1>
    <ul style={{ margin: 0, padding: 0, listStyle: 'none' }}>
      {jargons.map(jargon => (
        <li key={jargon.slug || ''} className="u-gap-bottom-xsmall">
          <Link to={`/${jargon.slug || ''}`} className="u-text-lowercase">
            {jargon.title}
          </Link>
        </li>
      ))}
    </ul>
  </Fragment>
);

export default TagDetail;
