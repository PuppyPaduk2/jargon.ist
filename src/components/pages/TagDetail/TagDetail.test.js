// @flow

import React from 'react';
import { shallow } from 'enzyme';
import Link from 'gatsby-link';

import jargonsFixture from '../../../../__fixtures__/jargons';

import TagDetail from './TagDetail';

it('renders correctly', () => {
  const wrapper = shallow(<TagDetail tag="tag1" totalCount={3} jargons={jargonsFixture} />);

  expect(wrapper).toMatchSnapshot();
});

it('renders link to jargon details', () => {
  const wrapper = shallow(<TagDetail tag="tag1" totalCount={3} jargons={jargonsFixture} />);

  jargonsFixture.forEach(jargon => {
    expect(wrapper.find(Link).find(`[to='/${jargon.slug}']`)).toHaveLength(1);
  });
});
