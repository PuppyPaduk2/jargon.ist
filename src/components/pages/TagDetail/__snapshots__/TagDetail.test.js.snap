// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<React.Fragment>
  <h1>
    tag1
     (
    3
    )
  </h1>
  <ul
    style={
      Object {
        "listStyle": "none",
        "margin": 0,
        "padding": 0,
      }
    }
  >
    <li
      className="u-gap-bottom-xsmall"
      key="example-jargon-1"
    >
      <GatsbyLink
        className="u-text-lowercase"
        to="/example-jargon-1"
      >
        Example Jargon 1
      </GatsbyLink>
    </li>
    <li
      className="u-gap-bottom-xsmall"
      key="example-jargon-2"
    >
      <GatsbyLink
        className="u-text-lowercase"
        to="/example-jargon-2"
      >
        Example Jargon 2
      </GatsbyLink>
    </li>
  </ul>
</React.Fragment>
`;
